bricks/distributed/vector_clock.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include "../../bricks/time/chrono.h"
+
+class DiscreteClocks : public std::vector<uint64_t> {
```

Comment:
FWIW, it's strictly discouraged to inherit from SQL containers.

The recommended solution is has-a, not is-a.

If you want it to be generic, might as well add some `: public ActsAsVector<uint64_t>`, which would contain all the necessary methods (`begin()`, `end()`, `size()`, `empty()`, and `operator[]` are enough, although you'd probably need `.resize()` and `.reserve()` too), and contain the very `vector`, so that there's a simple way to access it, perhaps even with a cast.

Reply:
@dkorolev SQL containers? 😁

Reply:
LOL

No =)